#VRML_SIM R2022b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/factory/containers/protos/WoodenBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2022b/projects/objects/floors/protos/Floor.proto"

WorldInfo {
  contactProperties [
    ContactProperties {
      material2 "db"
      coulombFriction [
        0.5
      ]
    }
  ]
}
Viewpoint {
  orientation 0.3864336977363179 0.6553321224494917 -0.6490060142553999 1.3624144330525643
  position -1.2258591214264491 2.261294830106828 3.767355545313794
}
TexturedBackground {
}
TexturedBackgroundLight {
}
WoodenBox {
  translation -0.28022002603346 0.40745693963150054 0.049607599999993285
  rotation -3.6438679028697364e-14 -1.1675429931999896e-14 -1 1.4360145548263612
  size 0.1 0.1 0.1
  mass 0.4
}
WoodenBox {
  translation 0.289801 0.716234 0.0496076
  rotation -2.1763746328224018e-07 -2.2573219862503316e-07 -0.9999999999999509 1.5593174418210558
  name "wooden box(1)"
  size 0.1 0.1 0.1
  mass 1
}
WoodenBox {
  translation -0.7357550000023427 0.0011082899945671775 0.04988216898434828
  rotation -3.523201416885393e-14 -1.1375681686776461e-14 -1 1.4360145548263612
  name "wooden box(2)"
  size 0.1 0.1 0.1
  mass 0.3
}
WoodenBox {
  translation -0.0759135 -0.411183 0.0496076
  rotation -3.649533543097892e-14 -1.1630272067062146e-14 -1 1.4360145548263612
  name "wooden box(3)"
  size 0.1 0.1 0.1
  mass 1
}
WoodenBox {
  translation 0.4599399977259038 -0.2902940052733591 0.0496076
  rotation -3.6464932587390713e-14 -1.1724177258985107e-14 -1 1.4360145548263612
  name "wooden box(4)"
  size 0.1 0.1 0.1
  mass 0.4
}
WoodenBox {
  translation -0.370958 -0.680265 0.0496076
  rotation -3.64624637414861e-14 -1.1687602539975182e-14 -1 1.4360145548263612
  name "wooden box(5)"
  size 0.1 0.1 0.1
  mass 1
}
WoodenBox {
  translation 0.728537 0.71432 0.0498038
  rotation -3.4919146384063375e-14 -1.1208499493995034e-14 -1 1.4360145548263612
  name "wooden box(6)"
  size 0.1 0.1 0.1
  mass 0.5
}
WoodenBox {
  translation 0.0145809 -0.00524774 0.0496072
  rotation -0.0025906511063785278 -0.000659079281470231 0.9999964270642898 -1.6257853071795862
  name "wooden box(7)"
  size 0.1 0.1 0.1
  mass 1
}
WoodenBox {
  translation 0.6708479739665403 0.21618893963150043 0.049607599999993306
  rotation -3.643318114713396e-14 -1.1662022978146978e-14 -1 1.4360145548263612
  name "wooden box(8)"
  size 0.1 0.1 0.1
  mass 0.4
}
Floor {
  size 2 2
  tileSize 0.25 0.25
}
Robot {
  translation 0.390945 0.385372 0.0397125
  rotation -0.000510746786035042 0.0016052993274988457 0.9999985810748883 2.00599
  children [
    DEF Body Shape {
      appearance PBRAppearance {
        baseColor 1 1 0
        metalness 0
      }
      geometry Box {
        size 0.2 0.1 0.05
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 258.79011386534086
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "Wheel1"
        }
      ]
      endPoint Solid {
        translation 0.0500017780623101 0.06000000009977072 3.5792689507358633e-06
        rotation 0.7261672846550857 0.48634220680821505 -0.48595507258755405 1.884689559827607
        children [
          DEF Wheel Shape {
            appearance PBRAppearance {
              baseColor 0.666667 0.333333 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE Wheel
        physics DEF Physics_wh Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 190.30778042904814
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "Wheel2"
        }
      ]
      endPoint Solid {
        translation 0.05001178115317744 -0.05999991723334277 -3.865896414139686e-06
        rotation -0.48453384218411133 -0.6188039413216752 0.6183111174668168 4.044614687135764
        children [
          USE Wheel
        ]
        name "solid(1)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 259.6137782261691
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "Wheel3"
        }
      ]
      endPoint Solid {
        translation -0.04998850207264452 0.06000011177327655 -2.6694116925292824e-06
        rotation 0.41207730000957454 0.6445359000380593 -0.6440231147862225 2.359283161410289
        children [
          USE Wheel
        ]
        name "solid(2)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 190.45569377493123
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "Wheel4"
        }
      ]
      endPoint Solid {
        translation -0.049996706298812754 -0.05999999990128034 -2.277999921572851e-06
        rotation -0.42832343540003137 -0.6392140741989708 0.6387052544273201 3.9515337436098856
        children [
          USE Wheel
        ]
        name "solid(3)"
        boundingObject USE Wheel
        physics USE Physics_wh
      }
    }
    DistanceSensor {
      translation 0.1 0.02 0
      rotation 0 0 -1 -0.038205307179586256
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_left"
      boundingObject USE sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.6 1000 0
      ]
      aperture 1.57
    }
    DistanceSensor {
      translation 0.1 -0.02 0
      rotation 0 0 -1 0.00820061
      children [
        DEF sensor Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "ds_right"
      boundingObject USE sensor
      physics Physics {
      }
      lookupTable [
        0 0 0
        0.6 1000 0
      ]
    }
    DistanceSensor {
      translation 0.085 0 -0.023
      rotation -0.004100340762836766 -0.9999831860391573 0.004100590762883277 -1.5708153071795863
      children [
        DEF colorSensor Shape {
          appearance PBRAppearance {
            baseColor 0.333333 1 1
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "colorSensor"
      boundingObject USE colorSensor
      physics Physics {
      }
      type "infra-red"
    }
  ]
  boundingObject USE Body
  physics Physics {
    density -1
    mass 1
  }
  controller "sumo"
}
